src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -290,6 +306,22 @@ void configSoftKernel(xclSKCmd *cmd)
   act.sa_handler = sigAct;
   sigemptyset(&act.sa_mask);
   act.sa_flags = 0;
+  sigaction(SIGHUP , &act, 0);
+  sigaction(SIGINT, &act, 0);
+  sigaction(SIGQUIT , &act, 0);
+  sigaction(SIGILL, &act, 0);
+  sigaction(SIGTRAP, &act, 0);
+  sigaction(SIGABRT, &act, 0);
+  sigaction(SIGBUS, &act, 0);
+  sigaction(SIGFPE, &act, 0);
+  sigaction(SIGKILL, &act, 0);
+  sigaction(SIGUSR1, &act, 0);
+  sigaction(SIGSEGV, &act, 0);
+  sigaction(SIGUSR2, &act, 0);
+  sigaction(SIGPIPE, &act, 0);
+  sigaction(SIGALRM, &act, 0);
+  sigaction(SIGTERM, &act, 0);
+
```

Comment:
If I read the code correctly, are these signal handlers registered in the parent process? I think we really want to capture those signal in the softkernel process itself, right? When softkernel process crash, will it send, say SIGSEGV, to the parent process?

Reply:
The installation of the handlers happens in the parent, and should/will carry forward to the child. In some sense this is better, since the original skd(parent) and the rest of the soft-kernels (children) will be logging backtrace incase of signals.

Reply:
By that I mean, after the fork() the same handling will be installed on parent and child processes.

Reply:
Good to know it is inherited by the child process.

Then I have a question about wait call in the child process. It does not make sense to me to call wait in the child process. I think we should call exit explicitly there.

Reply:
My suggestion is to register different sig handler in parent and child process separately, where you don't call wait in child's handler.